src/database-controller/src/core/framework.js

```diff
@@ -0,0 +1,421 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+const logger = require('@dbc/core/logger');
+const k8s = require('@dbc/core/k8s');
+const _ = require('lodash');
+const yaml = require('js-yaml');
+
+const mockFrameworkStatus = () => {
+  return {
+    state: 'AttemptCreationPending',
+    attemptStatus: {
+      completionStatus: null,
+      taskRoleStatuses: [],
+    },
+    retryPolicyStatus: {
+      retryDelaySec: null,
+      totalRetriedCount: 0,
+      accountableRetriedCount: 0,
+    },
+  };
+};
+
+const convertState = (state, exitCode, retryDelaySec) => {
+  switch (state) {
+    case 'AttemptCreationPending':
+    case 'AttemptCreationRequested':
+    case 'AttemptPreparing':
+      return 'WAITING';
+    case 'AttemptRunning':
+      return 'RUNNING';
+    case 'AttemptDeletionPending':
+    case 'AttemptDeletionRequested':
+    case 'AttemptDeleting':
+      if (exitCode === -210 || exitCode === -220) {
+        return 'STOPPING';
+      } else {
+        return 'RUNNING';
+      }
+    case 'AttemptCompleted':
+      if (retryDelaySec == null) {
+        return 'RUNNING';
+      } else {
+        return 'WAITING';
+      }
+    case 'Completed':
+      if (exitCode === 0) {
+        return 'SUCCEEDED';
+      } else if (exitCode === -210 || exitCode === -220) {
+        return 'STOPPED';
+      } else {
+        return 'FAILED';
+      }
+    default:
+      return 'UNKNOWN';
+  }
+};
+
+function ignoreError(err) {
+  logger.info('This error will be ignored: ', err);
+}
+
+// Class `Snapshot` handles the full json of framework.
+// It provides method like:
+//    getRequest: extract framework request from the full json
+//    overrideRequest: override the framework request using another snapshot
+//    getRequestUpdate, getStatusUpdate, getAllUpdate: Get database updates from the snapshot.
```

Comment:
>getRequestUpdate, getStatusUpdate, getAllUpdate: Get database updates from the snapshot. [](start = 6, length = 88)

The comment is not clear. What is "database updates"?

Reply:
Commented.